### PR TITLE
logpolicy: add some docs

### PR DESCRIPTION
**Pull request by bradfitz**
And minor cleanups.

Let me know if my docs are accurate.

### CODE REVIEW

File: logpolicy/logpolicy.go [hunk 1]
```diff
@@ -2,6 +2,9 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package logpolicy manages the creation of reuse of logtail loggers,
```
Comment (danderson) [inline logpolicy/logpolicy.go]: Typo: ITYM "creation or reuse" ?

File: logpolicy/logpolicy.go [hunk 2]
```diff
@@ -40,54 +48,66 @@ func (c *Config) ToBytes() []byte {
...
+func (l logWriter) Write(buf []byte) (int, error) {
+	l.logger.Printf("%s", buf)
 	return len(buf), nil
 }
 
-func New(collection string, filePrefix string) *Policy {
-	statefile := filePrefix + ".log.conf"
+// New returns a new log policy (a logger and its instance ID) for a
+// given collection name.  The provided filePrefix is used as a prefix
+// as the filenames for both for the logger's state file, as well as
```
Comment (danderson) [inline logpolicy/logpolicy.go]: typo? "as a prefix as the filenames"

File: logpolicy/logpolicy.go [hunk 3]
```diff
@@ -40,54 +48,66 @@ func (c *Config) ToBytes() []byte {
 	return data
 }
 
-func (c *Config) Save(statefile string) {
+// Save writes the JSON representation of c to stateFile.
+func (c *Config) Save(stateFile string) error {
```
Comment (crawshaw) [inline logpolicy/logpolicy.go]: Do you have a separate PR to use this error so it isn't lost?

File: logpolicy/logpolicy.go (hunk 1 above)
Comment (bradfitz) [reply logpolicy/logpolicy.go]: Yup, thanks.

File: logpolicy/logpolicy.go (hunk 3 above)
Comment (bradfitz) [reply logpolicy/logpolicy.go]: The only caller I see is also updated in this PR already.

File: logpolicy/logpolicy.go (hunk 2 above)
Comment (bradfitz) [reply logpolicy/logpolicy.go]: Done.
